api/src/main/java/org/apache/iceberg/BlobMetadata.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
```

Comment:
We typically omit `NonNull` / `NotNull` / `Nullable` annotations since there are so many of them from various places.

Reply:
Some attributes are nullable and some are not. For numeric values I can denote it with using primitive or boxed type.
How should i indicate something may be null or that something shouldn't be null in other cases?

Reply:
Dropped anyway.